clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -308,14 +308,6 @@ static void reportConflictingAttrs(Sema &S, FunctionDecl *F, const Attr *A1,
   F->setInvalidDecl();
 }
 
-// Returns the signed constant integer value represented by given expression.
-static int64_t getIntExprValue(Sema &S, const Expr *E) {
```

Comment:
I decided to replace this function with extension to `IntelReqdSubGroupSizeAttr` class to add caching functionality to avoid re-evaluating argument expression each time we need the value, but I'm not entirely sure that this is a correct approach or I implemented it correctly - please advise here

Reply:
I have a feeling, that I've already seen that type of function is Sema, like 'check unsigned int arguments' (can't grep it right now), so I see no value in the function above.

Reply:
Should we use this function then?

Reply:
> Should we use this function then?

I prefer to have some cached value instead of evaluating attribute argument each time we need it to perform some correctness check